pkg/version/version.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"fmt"
+	"runtime"
+
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+)
+
+// Get returns the overall codebase version. It's for detecting
+// what code a binary was built from.
+func Get() apimachineryversion.Info {
```

Comment:
#646 is just dumping the version information to the CLI.  #646 should be updated to pull from this version struct to be honest.

for reference, every kube style api server supports a `/version` endpoint.

for example:

```
$ kubectl get --raw /version
{
  "major": "1",
  "minor": "7+",
  "gitVersion": "v1.7.0-alpha.1.408+c5cbecc4547e33-dirty",
  "gitCommit": "c5cbecc4547e33cb72dc3372043abfd22d6cbdce",
  "gitTreeState": "dirty",
  "buildDate": "2017-04-13T19:13:16Z",
  "goVersion": "go1.7.5",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```

right now, if i point my client to the same endpoint on service catalog api server:

```
{
  "major": "",
  "minor": "",
  "gitVersion": "v0.0.0-master+$Format:%h$",
  "gitCommit": "$Format:%H$",
  "gitTreeState": "not a git tree",
  "buildDate": "1970-01-01T00:00:00Z",
  "goVersion": "go1.7.5",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```

as a follow-on to this pr, we need to get git version, tree, commit and build date returning the proper values.